src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/SearchValueExpressionBuilderHelper.cs

```diff
@@ -295,24 +296,60 @@ void ISearchValueVisitor.Visit(UriSearchValue uri)
         {
             EnsureArg.IsNotNull(uri, nameof(uri));
 
-            switch (_modifier?.SearchModifierCode)
+            // Query should be "(FieldName.Uri == uri.ToString() and FieldName.UriVersion == null and FieldName.UriFragment == null) or
```

Comment:
Is this for compatibility with data that was previously written to the database?

Reply:
We always need to try to parse an incoming search as a Canonical because the search type is always Uri. If it looked like a canonical, but the model type said to store as a Uri or Url this should match.